src/polymer/call-super-in-callbacks.ts

```diff
@@ -0,0 +1,221 @@
+/**
+ * @license
+ * Copyright (c) 2017 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at
+ * http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at
+ * http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+import * as estraverse from 'estraverse';
+import * as estree from 'estree';
+import {Document, Element, ElementMixin, isPositionInsideRange, ParsedDocument, Severity, SourceRange, Warning} from 'polymer-analyzer';
+
+import {registry} from '../registry';
+import {Rule} from '../rule';
+import {stripWhitespace} from '../util';
+
+import stripIndent = require('strip-indent');
+
+const methodsThatMustCallSuper = new Set([
+  'ready',
+  'connectedCallback',
+  'disconnectedCallback',
+  'attributeChangedCallback',
+]);
+
+export class CallSuperInCallbacks extends Rule {
+  code = 'call-super-in-callbacks';
+  description = stripIndent(`
+      Warns when a Polymer element overrides one of the custom element callbacks
+      but does not call super.callbackName().
+  `).trim();
+
+  constructor() {
+    super();
+  }
+
+  async check(document: Document) {
+    const warnings: Warning[] = [];
+
+    for (const element of document.getByKind('polymer-element')) {
```

Comment:
Why is this polymer-element, and not element? And should this also contain element-mixin?